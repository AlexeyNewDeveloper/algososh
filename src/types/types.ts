export interface IItemObject {
  value: string;
  id: string;
  modified: boolean;
  changing: boolean;
}
