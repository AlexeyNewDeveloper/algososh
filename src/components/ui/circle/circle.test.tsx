import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Корректная отрисовка элемента Circle", () => {
  it("Отрисовка Circle без буквы", () => {
    const tree = renderer.create(<Circle letter="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Отрисовка Circle с буквами", () => {
    it("Одна буква", () => {
      const tree = renderer.create(<Circle letter="A" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Две буквы", () => {
      const tree = renderer.create(<Circle letter="AA" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Три буквы", () => {
      const tree = renderer.create(<Circle letter="AAA" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Четыре буквы", () => {
      const tree = renderer.create(<Circle letter="AAAA" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("Отрисовка Circle с head", () => {
    const tree = renderer.create(<Circle head="head" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle с React-элементом в качестве head", () => {
    const tree = renderer
      .create(<Circle head={<Circle letter="A" isSmall={true} />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle с tail", () => {
    const tree = renderer.create(<Circle tail="tail" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle с React-элементом в качестве tail", () => {
    const tree = renderer
      .create(<Circle tail={<Circle letter="A" isSmall={true} />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle с index", () => {
    const tree = renderer.create(<Circle index={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle с пропсом isSmall", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle в состоянии default", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle в состоянии changing", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка Circle в состоянии modified", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
