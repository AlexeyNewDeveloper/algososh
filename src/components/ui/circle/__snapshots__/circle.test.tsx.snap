// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Корректная отрисовка элемента Circle Отрисовка Circle без буквы 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle в состоянии changing 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   changing"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle в состоянии default 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle в состоянии modified 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   modified"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с React-элементом в качестве head 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  >
    <div
      className="content "
      data-testid="testingCircle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circleHead"
      />
      <div
        className="circle  small default"
        data-testid="circleColorState"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circleLetter"
        >
          A
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circleTail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с React-элементом в качестве tail 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  >
    <div
      className="content "
      data-testid="testingCircle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circleHead"
      />
      <div
        className="circle  small default"
        data-testid="circleColorState"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circleLetter"
        >
          A
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circleTail"
      />
    </div>
  </div>
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с head 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="circleHead"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с index 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с tail 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circleTail"
  >
    tail
  </div>
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с буквами Две буквы 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    >
      AA
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с буквами Одна буква 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с буквами Три буквы 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    >
      AAA
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с буквами Четыре буквы 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    >
      AAAA
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Корректная отрисовка элемента Circle Отрисовка Circle с пропсом isSmall 1`] = `
<div
  className="content "
  data-testid="testingCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle  small default"
    data-testid="circleColorState"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;
